desc/grid.py

```diff
@@ -187,17 +187,25 @@ def _scale_weights(self):
     @property
     def L(self):
         """int: Radial grid resolution."""
-        return self.__dict__.setdefault("_L", 0)
+        if getattr(self, "_L", None) is None:
+            self._L = self.num_rho - 1
+        return self._L
```

Comment:
i used convention for linear grid. quadrature and concentric grid always define these attributes, so they won't get overwritten.

if it's preferred to move this logic do lineargrid only by overriding these property functions, feel free to